Assets/LeapMotion/Core/Scripts/Hands/RiggedFinger.cs

```diff
@@ -12,40 +12,95 @@
 using Leap;
 
 namespace Leap.Unity {
-  /**
-* Manages the orientation of the bones in a model rigged for skeletal animation.
-* 
-* The class expects that the graphics model bones corresponding to bones in the Leap Motion 
-* hand model are in the same order in the bones array.
-*/
+
+  /// <summary>
+  /// Manages the position and orientation of the bones in a model rigged for skeletal
+  /// animation.
+  ///  
+  /// The class expects that the graphics model's bones that correspond to the bones in
+  /// the Leap Motion hand model are in the same order in the bones array.
+  /// </summary>
   public class RiggedFinger : FingerModel {
 
-    /** Allows the mesh to be stretched to align with finger joint positions
-     * Only set to true when mesh is not visible
-     */
+    /// <summary>
+    /// Allows the mesh to be stretched to align with finger joint positions.
+    /// Only set to true when mesh is not visible.
+    /// </summary>
     [HideInInspector]
     public bool deformPosition = false;
 
+    [HideInInspector]
+    public bool scaleLastFingerBone = false;
+
     public Vector3 modelFingerPointing = Vector3.forward;
     public Vector3 modelPalmFacing = -Vector3.up;
 
     public Quaternion Reorientation() {
       return Quaternion.Inverse(Quaternion.LookRotation(modelFingerPointing, -modelPalmFacing));
     }
 
-    private RiggedHand riggedHand;
+    /// <summary> Backing store for s_standardFingertipLengths, don't touch. </summary>
+    private static float[] s_backingStandardFingertipLengths = null;
+    static RiggedFinger() {
+      // Calculate standard fingertip lengths.
+      s_backingStandardFingertipLengths = new float[5];
+      var testHand = TestHandFactory.MakeTestHand(isLeft: true,
+                           unitType: TestHandFactory.UnitType.UnityUnits);
+      for (int i = 0; i < 5; i++) {
+        var fingertipBone = testHand.Fingers[i].bones[3];
+        s_backingStandardFingertipLengths[i] = fingertipBone.Length;
+      }
+    }
+    /// <summary>
+    /// Fingertip lengths for the standard edit-time hand.
+    /// </summary>
+    private static float[] s_standardFingertipLengths {
```

Comment:
Why the getter, can't you just use the array directly?

Reply:
woop sorry yeah can change